tests/46direct/01directmessage.pl

```diff
@@ -0,0 +1,143 @@
+use Future::Utils qw( repeat );
+
+my $next_device_message_txn_id = 0;
+
+push our @EXPORT, qw( matrix_send_to_device_message );
+
+sub matrix_send_to_device_message
+{
+   my ( $user, %params ) = @_;
+   exists $params{type} or die "Expected a type";
+   exists $params{messages} or die "Expected messages";
+   my $type = delete $params{type};
```

Comment:
Is `type` ever false? If not it might be more direct to

    my $type = delete $params{type} or die "Expected a 'type'";